openpype/hosts/maya/plugins/publish/validate_resolution.py

```diff
@@ -0,0 +1,54 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from maya import cmds
+from openpype.hosts.maya.api.lib import reset_scene_resolution
+
+
+class ValidateResolution(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+    """Validate the resolution setting aligned with DB"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    families = ["renderlayer"]
+    hosts = ["maya"]
+    label = "Validate Resolution"
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        width, height = self.get_db_resolution(instance)
+        current_width = cmds.getAttr("defaultResolution.width")
+        current_height = cmds.getAttr("defaultResolution.height")
```

Comment:
This is wrong - it should actually check the value **in** the renderlayer because there might be renderlayer overrides. Look at `lib.get_attr_in_layer`.

I believe the layer would be `layer = instance.data["renderlayer"]`
So this would be:
```python
layer = instance.data["renderlayer"]
current_width = lib.get_attr_in_layer("defaultResolution.width", layer=layer)
current_height = lib.get_attr_in_layer("defaultResolution.height", layer=layer)
```

Then in comes the next complication, because V-Ray doesn't use the width/height from `defaultResolution` but from its own `VRaySettingsNode`. Have a look at `lib_rendersettings.RenderSettings` class. Here's [the V-Ray part](https://github.com/ynput/OpenPype/blob/develop/openpype/hosts/maya/api/lib_rendersettings.py#L341-L342) setting the values.